src/app/CommandHandler.h

```diff
@@ -202,6 +202,8 @@ class CommandHandler
      */
     Messaging::ExchangeContext * GetExchangeContext() const { return mpExchangeCtx; }
 
+    NodeId SourceNodeId() const { return mpExchangeCtx->GetSessionHandle()->AsSecureSession()->GetPeerNodeId(); }
```

Comment:
The reason I brought up groups is that `AsSecureSession()` will _crash_ if this is a group session.

Group seessions may have a source node id, but this is not a way to get it...

@kghost

Reply:
Got it, added a check for group session and for now returning `kUndefinedNodeId`.